libevmasm/Assembly.cpp

```diff
@@ -74,6 +75,130 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::addAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solAssert(_code.isArray());
+	for (auto const& jsonItem: _code)
+		m_items.emplace_back(createAssemblyItemFromJSON(jsonItem));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// During the assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we just ignore exactly these `JUMPDEST`'s.
+		auto const nextItem = next(current);
```

Comment:
This code would be nicer to read with some types. These are all `Json::Value`, right?
```suggestion
	for (Json::Value const& jsonItem: _code)
		m_items.emplace_back(createAssemblyItemFromJSON(jsonItem));

	for (Json::Value current = m_items.begin(); current != m_items.end(); ++current)
	{
		// During the assembly json export a `JUMPDEST` is always generated after a `tag`.
		// So we just ignore exactly these `JUMPDEST`'s.
		Json::Value const nextItem = next(current);
```

Same below.

Reply:
I changed the first `for` loop to be `Json::Value`. For the other loops we have here iterators of type `std::vector<AssemblyItem>::iterator `. So I guess the confusing thing here was the naming of `nextItem`, changed to `nextIter` now.